controllers/kubeadmconfig_controller.go

```diff
@@ -330,27 +355,46 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *capiv1alpha2.Clust
 	return nil
 }
 
-func (r *KubeadmConfigReconciler) getCertificates() (cloudinit.Certificates, error) {
-	//TODO(fp) check what is the expected flow for certificates
-	certificates, _ := NewCertificates()
-
-	return cloudinit.Certificates{
-		CACert:           string(certificates.ClusterCA.Cert),
-		CAKey:            string(certificates.ClusterCA.Key),
-		EtcdCACert:       string(certificates.EtcdCA.Cert),
-		EtcdCAKey:        string(certificates.EtcdCA.Key),
-		FrontProxyCACert: string(certificates.FrontProxyCA.Cert),
-		FrontProxyCAKey:  string(certificates.FrontProxyCA.Key),
-		SaCert:           string(certificates.ServiceAccount.Cert),
-		SaKey:            string(certificates.ServiceAccount.Key),
-	}, nil
+func (r *KubeadmConfigReconciler) getClusterCertificates(ctx context.Context, clusterName, namespace string) (*certs.Certificates, error) {
+	secret := &corev1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-certs", clusterName), Namespace: namespace}, secret)
+	if err != nil {
+		return nil, err
+	}
+
+	return certs.MapToCertificates(secret.Data), nil
 }
 
-// SetupWithManager TODO
-func (r *KubeadmConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cabpkv1alpha2.KubeadmConfig{}).
-		Complete(r)
+func (r *KubeadmConfigReconciler) createClusterCertificates(ctx context.Context, clusterName string, config *cabpkv1alpha2.KubeadmConfig) (*certs.Certificates, error) {
+	certificates, err := certs.NewCertificates()
+	if err != nil {
+		return nil, err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-certs", clusterName),
+			Namespace: config.GetNamespace(),
+			OwnerReferences: []v1.OwnerReference{
+				{
+					// TODO: Consider whether this should have a cluster object as its owner
```

Comment:
I think it is fine to just have the KubeadmConfig as an owner, since this is directly related to this provider.